### ISSUE TITLE
triton: Documentation example incomplete 

### ISSUE THREAD

**Issue by rickard-von-essen** [hashicorp/packer#4814]
The example in the docs is not complete. 

```json
{
  "builders": [
    {
      "type": "triton",
      "triton_account": "triton_username",
      "triton_key_id": "6b:95:03:3d:d3:6e:52:69:01:96:1a:46:4a:8d:c1:7e",
      "triton_key_material": "~/.ssh/id_rsa",
      "source_machine_name": "image-builder",
      "source_machine_package": "g3-standard-0.5-smartos",
      "source_machine_image": "70e3ae72-96b6-11e6-9056-9737fd4d0764",
      "ssh_username": "root",
      "ssh_private_key_file": "~/.ssh/id_rsa",
      "image_name": "my_new_image",
      "image_version": "1.0.0",
  }
]
}
```

- `~` doesn't work.
- `source_machine_package` is outdated.

Fixing these result in:
```
Build 'triton' errored: Problem creating source machine: failed to create machine with name: image-builder
caused by: Missing parameters https://us-east-1.api.joyent.com/rickard.von.essen/machines
caused by: request "https://us-east-1.api.joyent.com/rickard.von.essen/machines" returned unexpected status 409 with body "{\"code\":\"ValidationFailed\",\"message\":\"Invalid VM parameters: Invalid networks\",\"errors\":[{\"field\":\"networks\",\"code\":\"Invalid\",\"message\":\"Cannot set property 'primary' of undefined\"}]}"
```

**Comment by mwhooker** [comment]
I'm going to be setting up a triton account today to test #4623 so I can flesh out the docs

**Comment by rickard-von-essen** [comment]
Ping @siepkes

**Comment by rickard-von-essen** [comment]
Exactly what I tried and failed.

**Comment by siepkes** [comment]
@rickard-von-essen Thought I tested that example but apparently I didn't or make some last minute untested changes. Using this should work:

```
"triton_key_material": ${file("~/.ssh/id_rsa")}
```

@mwhooker If you don't get around to it, let me know and i'll run over the Triton docs.

**Comment by mwhooker** [comment]
@siepkes thanks for following up. I probably won't be able to get to this until next week. PRs appreciated!

**Comment by siepkes** [comment]
@mwhooker No problem, I'll take a look at it!

**Comment by siepkes** [comment]
The following works on my private Triton instance. However I don't have access to the Joyent public cloud so I can't verify all of it:

```
{
  "builders": [
    {
      "type": "triton",
      "triton_url": "https://cloudapi.somewhere.nl",
      "triton_account": "siepkes",
      "triton_key_id": "10:de:8e:d1:8d:1b:40:27:1d:06:ee:07:93:8e:12:f4",
      "triton_key_material": "/home/siepkes/.ssh/id_rsa",
      "source_machine_name": "image-builder",
      "source_machine_package": "sp_normal_2048",
      "source_machine_image": "08589eda-d64e-11e6-89a3-0b55371be927",
      "source_machine_networks": "7a56a9ed-6e14-4d2b-8096-8893bb592d78",
      "ssh_username": "root",
      "ssh_private_key_file": "/home/siepkes/.ssh/id_rsa",
      "image_name": "my_new_image",
      "image_version": "1.0.0"
    }
  ]
}
```

`triton_url` will be removed since it's only needed with a private Triton instance.

Could @jen20 or @mwhooker provide me with valid / validate values for the keys below? Thanks in advance!

* `source_machine_networks` Should be the default Joyent external network. I don't know what the value should be.
* `source_machine_package` I think this can be `g3-standard-0.5-smartos`, is that correct?
* `source_machine_image` I assume the image `08589eda-d64e-11e6-89a3-0b55371be927`  (base 64 LTS) is present on the Joyent public cloud?

As for `source_machine_package` being outdated; I need to look in to what replaces it.

**Comment by rickard-von-essen** [comment]
```shell
$ triton pkg ls

SHORTID   NAME                    MEMORY  SWAP  DISK  VCPUS
14ad9d54  g4-highcpu-128M           128M  512M    3G      -
14ae2634  g4-highcpu-256M           256M    1G    5G      -
14aea8fc  g4-highcpu-512M           512M    2G   10G      -
14af2214  g4-highcpu-1G               1G    4G   25G      -
14afab8a  g4-highcpu-2G               2G    8G   50G      -
14b0351e  g4-highcpu-4G               4G   16G  100G      -
14b0c254  g4-highcpu-8G               8G   32G  200G      -
14b143dc  g4-highcpu-16G             16G   64G  400G      -
14b1c096  g4-highcpu-32G             32G  128G  800G      -
14b23918  g4-highcpu-64G             64G  128G  1.5T      -
14b4ff36  k4-highcpu-kvm-250M       256M    1G   10G      1
14b5760a  k4-highcpu-kvm-750M       768M    3G   25G      1
14b5edc4  k4-highcpu-kvm-1.75G      1.7G    7G   50G      1
14b67ef6  k4-highcpu-kvm-3.75G      3.7G   15G  100G      2
14b6fade  k4-highcpu-kvm-7.75G      7.7G   31G  200G      4
14b783d2  k4-highcpu-kvm-15.75G    15.7G   63G  400G      8
14a971a2  g4-general-4G               4G   16G   50G      -
14aa013a  g4-general-8G               8G   32G  100G      -
14aa8ed4  g4-general-16G             16G   64G  200G      -
14ab21c8  g4-general-32G             32G  128G  400G      -
14aba044  k4-general-kvm-3.75G      3.7G   15G   50G      1
14ac17a4  k4-general-kvm-7.75G      7.7G   31G  100G      2
14ac8f5e  k4-general-kvm-15.75G    15.7G   63G  200G      4
14ad1a32  k4-general-kvm-31.75G    31.7G  127G  400G      8
14b801c2  g4-highram-16G             16G   64G  100G      -
14b877c4  g4-highram-32G             32G  128G  200G      -
14b900cc  g4-highram-64G             64G  128G  400G      -
14b97ebc  g4-highram-110G           110G  220G  800G      -
14ba876c  k4-highram-kvm-15.75G    15.7G   63G  100G      2
14bafb20  k4-highram-kvm-31.75G    31.7G  127G  200G      4
14bb84f0  k4-highram-kvm-63.75G    63.7G  255G  400G      8
f5e17c0c  g4-bigdisk-16G             16G   64G  1.2T      -
14be9b2c  g4-bigdisk-32G             32G  128G  2.4T      -
14bc0524  g4-fastdisk-32G            32G  128G  800G      -
14bc8e86  g4-fastdisk-64G            64G  128G  1.5T      -
14bf1b9c  g4-bigdisk-64G             64G  128G  4.8T      -
14bd0f64  g4-fastdisk-110G          110G  220G  3.1T      -
14bf90cc  g4-bigdisk-110G           110G  220G  9.7T      -
7741b8f6  k4-bigdisk-kvm-15.75G    15.7G   63G  1.1T      2
14bd9600  k4-fastdisk-kvm-31.75G   31.7G  127G  800G      4
14c01a1a  k4-bigdisk-kvm-31.75G    31.7G  127G  2.4T      4
14be13c8  k4-fastdisk-kvm-63.75G   63.7G  255G  1.5T      8
14c0992c  k4-bigdisk-kvm-63.75G    63.7G  255G  4.8T      8
```

**Comment by rickard-von-essen** [comment]
```shell
$ triton image get f6acf198
{
    "id": "f6acf198-2037-11e7-8863-8fdd4ce58b6a",
    "name": "base-64",
    "version": "17.1.0",
    "os": "smartos",
    "requirements": {},
    "type": "zone-dataset",
    "description": "A 64-bit SmartOS image with just essential packages installed. Ideal for users who are comfortable with setting up their own environment and tools.",
    "files": [
        {
            "compression": "gzip",
            "sha1": "088f2c15e434cc86869190cc9137a97a79ccd7c1",
            "size": 184526338
        }
    ],
    "tags": {
        "role": "os",
        "group": "base-64"
    },
    "homepage": "https://docs.joyent.com/images/smartos/base",
    "published_at": "2017-04-13T10:57:19Z",
    "owner": "9dce1460-0c4c-4417-ab8b-25ca478c5a78",
    "public": true,
    "state": "active"
}
```

**Comment by rickard-von-essen** [comment]
```shell
$ triton network get f7ed95d3
{
    "id": "f7ed95d3-faaf-43ef-9346-15644403b963",
    "name": "Joyent-SDC-Public",
    "public": true
}
```

**Comment by rickard-von-essen** [comment]
```json
{
  "builders": [
    {
      "type": "triton",
      "triton_account": "rickard.von.essen",
      "triton_key_id": "90:30:b4:f8:c7:d6:7b:b9:f7:e1:8b:43:bd:10:17:8a",
      "triton_key_material": "/Users/rickard/.ssh/joyent_id_rsa",
      "source_machine_package": "g4-highcpu-512M",
      "source_machine_image": "70e3ae72-96b6-11e6-9056-9737fd4d0764",
      "source_machine_networks": [ "f7ed95d3-faaf-43ef-9346-15644403b963" ],
      "ssh_username": "root",
      "ssh_private_key_file": "/Users/rickard/.ssh/joyent_id_rsa",
      "image_name": "my_new_image",
      "image_version": "1.0.0"
    }
  ]
}
```

```shell
$ packer build -only triton triton.json
triton output will be in this color.

==> triton: Creating source machine...
Build 'triton' errored: Problem creating source machine: failed to create machine with name: packer-builder-my_new_image
caused by: Missing parameters https://us-east-1.api.joyent.com/rickard.von.essen/machines
caused by: request "https://us-east-1.api.joyent.com/rickard.von.essen/machines" returned unexpected status 409 with body "{\"code\":\"InvalidArgument\",\"message\":\"Invalid Networks\"}"

==> Some builds didn't complete successfully and had errors:
--> triton: Problem creating source machine: failed to create machine with name: packer-builder-my_new_image
caused by: Missing parameters https://us-east-1.api.joyent.com/rickard.von.essen/machines
caused by: request "https://us-east-1.api.joyent.com/rickard.von.essen/machines" returned unexpectedstatus 409 with body "{\"code\":\"InvalidArgument\",\"message\":\"Invalid Networks\"}"

==> Builds finished but no artifacts were created.
```

**Comment by siepkes** [comment]
@rickard-von-essen Weird. I'm not encountering any issues when I use my private Triton / Cloud API instance. Is there is anything special about that network or perhaps your account you can think of why this shouldn't work?

**Comment by rickard-von-essen** [comment]
I've basically only:

1) Registered
2) Setup an account (ssh key)
3) Installed and verified the `triton` cli works and can create VM's
4) Tried to use Packer with the same account and settings. Did not succeed because of the illegal arguments _invalid networks_.

**Comment by rickard-von-essen** [comment]
Ok, after adding some debug messages and reading the code, this works:

```json
{
  "builders": [
    {
      "type": "triton",
      "triton_url": "https://us-sw-1.api.joyentcloud.com",
      "triton_account": "rickard.von.essen",
      "triton_key_id": "90:30:b4:f8:c7:d6:7b:b9:f7:e1:8b:43:bd:10:17:8a",
      "triton_key_material": "/Users/rickard/.ssh/joyent_id_rsa",
      "source_machine_package": "g4-highcpu-512M",
      "source_machine_image": "70e3ae72-96b6-11e6-9056-9737fd4d0764",
      "source_machine_networks": [
        "1fc62c97-c1f0-41c6-9ef7-1f7ebe0ff09a",
        "f7ed95d3-faaf-43ef-9346-15644403b963"
      ],
      "ssh_username": "root",
      "ssh_private_key_file": "/Users/rickard/.ssh/joyent_id_rsa",
      "image_name": "my_new_image",
      "image_version": "1.0.0"
    }
  ]
}
```

The threw me off a bit was that in [triton/access_config.go#L30](https://github.com/hashicorp/packer/blob/master/builder/triton/access_config.go#L30), it defaults to `us-east-1` as default _datacenter_ even though this is not documented and all Joyent docs use `us-sw-1` as default _datacenter_.

I really think we should change the default to be what users expect (and clearly document it).

I'll do a PR which updates the docs and default `triton_url`.

**Comment by rickard-von-essen** [comment]
Closed in d2ae36214a84111314e053e527bca215a8e25b54
